File: problems/2073_time_needed_to_buy_tickets.py
class Solution:
    def timeRequiredToBuy(self, tickets: list[int], k: int) -> int:
        n = len(tickets)
        k_num = tickets[k]
        total = 0
        for i in range(n):
            if i < k:
                total += min(k_num, tickets[i])
            elif i == k:
                total += k_num
            else:
                total += min(k_num - 1, tickets[i])
        return total


print(Solution().timeRequiredToBuy([2, 3, 2], 2))
print(Solution().timeRequiredToBuy([5, 1, 1, 1], 0))
print(Solution().timeRequiredToBuy([84, 49, 5, 24, 70, 77, 87, 8], 3))

File: problems/2073-time-needed-to-buy-tickets.ts
function timeRequiredToBuy(tickets: number[], k: number) {
  let total = 0
  const kTime = tickets[k]
  for (let i = 0; i < tickets.length; i++) {
    if (i < k) {
      total += Math.min(tickets[i], kTime)
    } else if (i === k) {
      total += kTime
    } else {
      total += Math.min(kTime - 1, tickets[i])
    }
  }
  return total
}
